provider/pkg/gen/schema.go

```diff
@@ -829,6 +867,16 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 		}
 	}
 
+	if resourceSpec.Methods == nil {
+		resourceSpec.Methods = map[string]string{}
+	}
```

Comment:
Any reason not to unconditionally initialize this map when `pkg` is declared?

Reply:
Nope! Good catch. Fixed.